lib/saml/settings_service.rb

```diff
@@ -1,28 +1,43 @@
 # frozen_string_literal: true
 require 'memoist'
+require 'sentry_logging'
 
 module SAML
   # This class is responsible for putting together a complete ruby-saml
   # SETTINGS object, meaning, our static SP settings + the IDP settings
   # which must be fetched once and only once via IDP metadata.
   class SettingsService
     class << self
+      include SentryLogging
       extend Memoist
 
       METADATA_RETRIES = 3
       OPEN_TIMEOUT = 2
       TIMEOUT = 15
 
       def saml_settings
+        if @fetch_attempted.nil? || metadata_successfully_retrieved?
+          merged_saml_settings
+        else
+          # retrieve was attempted but failed, blow away memoized data and try again
+          merged_saml_settings(true)
```

Comment:
The named argument part would be helpful.  As it is now, one has to be familiar with the `memoist` gem,

As I read your 2nd comment it looks like it would undermine what the PR is trying to do, that is, add intelligence to `SettingsService` to know when it should return a memoized or brand new RubySaml::Settings object via a network GET call.  Would mind providing a pseudo code sample?